### PR TITLE
`Surface` will be created again after the front and back switches

### PR DESCRIPTION
**Pull request by Nayuta403**
fix：https://github.com/flutter/flutter/issues/105203

before:
https://user-images.githubusercontent.com/40540394/188312130-125265c2-f007-4f6f-83a8-330dae04d067.mp4

after:
https://user-images.githubusercontent.com/40540394/188312124-a69786ba-c825-4d11-b411-fa2a2758e85c.mp4

After calling `flutterView.setVisibility(View.GONE)`, switching from the background to the foreground.  Flutter will still be displayed. For `FlutterSurfaceView`, the Surface will be created again.

image

image

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Nayuta403** [comment]
cc @ColdPaleLight

**Comment by ColdPaleLight** [comment]
I'd like to defer to @dnfield  to take a look at it.

**Review by dnfield — APPROVED** [review]

**Comment by dnfield** [comment]
This will need a second review, maybe @GaryQian or @camsim99 ?

File: shell/platform/android/test/io/flutter/embedding/android/FlutterActivityAndFragmentDelegateTest.java
Comment (camsim99) [inline shell/platform/android/test/io/flutter/embedding/android/FlutterActivityAndFragmentDelegateTest.java:1065]: Very small nit:
```suggestion
  public void flutterSurfaceViewVisibilityChangedWithFlutterView() {
```

**Review by camsim99 — APPROVED** [review]
LGTM! Just one small nit!

File: shell/platform/android/test/io/flutter/embedding/android/FlutterActivityAndFragmentDelegateTest.java
Comment (Nayuta403) [reply shell/platform/android/test/io/flutter/embedding/android/FlutterActivityAndFragmentDelegateTest.java]: thanks~ done

**Pull request merged**
